### PR TITLE
refactor: SectionHolder

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/data/forms/dataentry/DataEntryAdapter.java [hunk 1]
```diff
@@ -54,23 +51,16 @@
     public DataEntryAdapter() {
         super(new DataEntryDiff());
         setHasStableIds(true);
-        sectionProcessor = PublishProcessor.create();
-        selectedSection = new ObservableField<>("");
         this.currentFocusUid = new MutableLiveData<>();
-        this.formViewHolderFactory = new FormViewHolderFactory(
-                sectionProcessor,
-                selectedSection);
+        this.formViewHolderFactory = new FormViewHolderFactory();
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/DataEntryAdapter.java:55]: :clap:

File: app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt [hunk 2]
```diff
@@ -64,14 +63,6 @@ class FormView @JvmOverloads constructor(
...
     // TODO methods to remove
 
-    fun sectionFlowable(): Flowable<String> {
-        return adapter.sectionFlowable()
-    }
-
-    fun setCurrentSection(selectedSection: String) {
-        adapter.setCurrentSection(selectedSection)
-    }
-
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:74]: :heart_eyes:

**Review by JaimeToca — APPROVED** [review]

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/edittext/EditTextViewModel.java [hunk 3]
```diff
@@ -110,4 +108,8 @@ public int getLayoutId() {
     public void onTextFilled(String value) {
         processor().onNext(RowAction.create(uid(), value, getAdapterPosition()));
     }
+
+    public boolean isLongText(){
+        return valueType() == ValueType.LONG_TEXT;
+    }
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/edittext/EditTextViewModel.java:114]: marvellous!

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt [hunk 4]
```diff
@@ -1,140 +1,24 @@
...
-    }
-
-    fun setSectionNumber(sectionNumber: Int) {
-        formBinding.sectionNumber.apply {
-            text = sectionNumber.toString()
-            background =
-                ContextCompat.getDrawable(itemView.context, R.drawable.ic_circle)
-        }
+    fun getSectionView(): SectionView {
+        return formBinding.sectionView
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt:22]: in order to remove all the holders, could we think to remove this from here. Let's check it when everything is merged. Good job with this class :clap:

**Review by andresmr — APPROVED** [review]

**Pull request merged**